### DISCUSSION TITLE
Fix defines all over the place.

### DISCUSSION THREAD

**Discussion by blockhead100** [NVIDIA/open-gpu-kernel-modules#268] [Ideas]
There are lots of #define all over like in nv-gpu-info.h where there is a #define NV_MAX_DEVICES 32 and in nvlimits.h it defines it again.
Should this be fixed?

![Screenshot from 2022-05-27 09-16-06](https://user-images.githubusercontent.com/55167737/170739768-6a54e9c0-50a8-4a7a-a50e-52a72651b67c.png)

![Screenshot from 2022-05-27 09-18-05](https://user-images.githubusercontent.com/55167737/170739787-695acd9f-20da-4b1e-9f59-08943996515e.png)

**Comment by aritger** [comment]
In general, nvlimits.h is intended to be the source of truth: it should have no dependencies on anything else.

Sure, it seems reasonable to me to delete NV_MAX_GPUs and have its current users include nvlimits.h and use NV_MAX_DEVICES instead.

Do you have other examples?  We probably should handle on a case-by-case basis.
